src/main/scala-3/com/raquo/airstream/split/SplittableTypeMacros.scala

```diff
@@ -0,0 +1,227 @@
+    pfListExpr: Expr[List[PartialFunction[Any, Any]]]
+  )(
+    using quotes: Quotes
+  ): List[Expr[PartialFunction[Any, Any]]] = {
+    import quotes.reflect.*
+    
+    pfListExpr match {
+      case '{ $headExpr :: (${tailExpr}: List[PartialFunction[Any, Any]]) } =>
+        headExpr :: exprOfListToListOfExpr(tailExpr)
+      case '{ Nil } => Nil
+      case _ => report.errorAndAbort("Macro expansion failed, please use `handleCase` instead of modify MatchSplitObservable explicitly")
+    }
+    
+  }
+
+  private def listOfExprToExprOfList(
+    pfExprList: List[Expr[PartialFunction[Any, Any]]]
+  )(
+    using quotes: Quotes
+  ): Expr[List[PartialFunction[Any, Any]]] = {
+    import quotes.reflect.*
+    
+    pfExprList match
+      case head :: tail => '{ $head :: ${listOfExprToExprOfList(tail)} }
+      case Nil => '{ Nil }
+  }
+
+  private inline def toSplittableOneObservable[Self[+_] <: Observable[_], O](
+    parentObservable: BaseObservable[Self, (Int, Any)],
+    handlerMap: Map[Int, Function[Any, O]]
+  ): Self[O] = {
+    parentObservable.matchStreamOrSignal(
+      ifStream = _.splitOne(_._1) { case (idx, (_, b), dataSignal) =>
+        val bSignal = dataSignal.map(_._2)
+        handlerMap.apply(idx).apply(b -> bSignal)
+      },
+      ifSignal = _.splitOne(_._1) { case (idx, (_, b), dataSignal) =>
+        val bSignal = dataSignal.map(_._2)
+        handlerMap.apply(idx).apply(b -> bSignal)
+      }
+    ).asInstanceOf[Self[O]] // #TODO[Integrity] Same as FlatMap/AsyncStatusObservable, how to type this properly?
+  }
+
+  private def observableImpl[Self[+_] <: Observable[_] : Type, I: Type, O: Type](
+    matchSplitObservableExpr: Expr[MatchSplitObservable[Self, I, O]]
+  )(
+    using quotes: Quotes
+  ): Expr[Self[O]] = {
+    import quotes.reflect.*
+
+    matchSplitObservableExpr match {
+      case '{ MatchSplitObservable.build[Self, I, O]($_, Nil, $_)} =>
+        report.errorAndAbort("Macro expansion failed, need at least one handleCase")
+      case '{ MatchSplitObservable.build[Self, I, O]($observableExpr, $caseListExpr, $handlerMapExpr)} => 
+        '{toSplittableOneObservable(
+          $observableExpr
+            .map(i => ${ innerObservableImpl('i, caseListExpr) })
+            .asInstanceOf[BaseObservable[Self, (Int, Any)]],
+          $handlerMapExpr
+        )}
+      case _ => report.errorAndAbort("Macro expansion failed, please use `splitMatch` instead of creating new MatchSplitObservable explicitly")
+    }
+  }
+
+  private def innerObservableImpl[I: Type](
+    iExpr: Expr[I],
+    caseListExpr: Expr[List[PartialFunction[Any, Any]]]
+  )(
+    using quotes: Quotes
+  ): Expr[(Int, Any)] = {
+    import quotes.reflect.*
+
+    val caseExprList = exprOfListToListOfExpr(caseListExpr)
+
+    val allCaseDefLists = caseExprList.reverse.zipWithIndex.flatMap { case (caseExpr, idx) =>
+      caseExpr.asTerm match {
+        case Block(List(DefDef(_, _, _, Some(Match(_, caseDefList)))), _) => {
```

Comment:
```suggestion
        case Lambda(_, Match(_, caseDefList)) => {
```

There's actually this little helper built into the `reflectModule` that matches the (awkwardly verbose) AST for anonymous functions. 👍 Awesome macro, btw!